hrqb/tasks/employee_leave.py

```diff
@@ -44,17 +46,28 @@ def get_dataframe(self) -> pd.DataFrame:
         dw_leaves_df = self.named_inputs["ExtractDWEmployeeLeave"].read()
         qb_emp_appts_df = self.named_inputs["ExtractQBEmployeeAppointments"].read()
 
-        qb_emp_appts_df = qb_emp_appts_df[["HR Appointment Key", "Record ID#"]].rename(
+        # join Employee Appointments from QB to get QB Record ID
+        dw_leaves_df = normalize_dataframe_dates(
+            dw_leaves_df,
+            ["appt_begin_date", "appt_end_date", "absence_date"],
+        )
+        dw_leaves_df["emp_appt_merge_key"] = dw_leaves_df.apply(
+            lambda row: TransformEmployeeAppointments.generate_merge_key(
```

Comment:
[Non-blocking] Hmm, is there a reason why `generate_merge_key` needs to be a static method of the `TransformEmployeeAppointments` class as opposed to a standalone util method? 🤔 Later in the module, a key is created for `leaves_df`--should `TransformEmployeeLeave` get its own static method `generate_merge_key`? 🤔

Reply:
The reasoning here is kind of twofold.

The logic for generating a merge field key on `TransformEmployeeAppointments` is needed by other tasks, but it's specific to how _that_ task does it, so keeping it associated with that task felt like it made sense.

Related -- at least not yet -- the other tasks that generate a merge, no _other_ tasks need to know how they do it.  If that changes, then I think a natural refactor would be to drop it into an externally available `@staticmethod`.

If this pattern continues where tasks need to be able to recreate the merge key for _other_ tasks, it might be worth abstracting that out to the base tasks, and then each task just identifies what row data is used for it.  But I don't think we're quite there yet.  This could be the end of this need.... or not!